### ISSUE TITLE
ESM broken

### ISSUE THREAD

**Issue by thantos** [cantoo-scribe/pdf-lib#29]
### What were you trying to do?

Use pdf-lib in node 18 with esm, esbuild (esm) to bundle, and run locally with nodejs in esm.

### How did you attempt to do it?

```ts
import {PDFDocument} from "@cantoo/pdf-lib"
```

```ts
import PdfLib from "@cantoo/pdf-lib"
const {PDFDocument} = PdfLib;
```

```ts
import * as  PdfLib from "@cantoo/pdf-lib"
```

### What actually happened?

Outcome of the import patterns in order

1. node 18 complains that the module is CJS and destructuring isn't supported
2. works for node, but esbuild complains that there is no default export for es/index.js
3. works for esbuild, but at runtime with esm on nodejs, all of the exports are undefined (because there is no default export).

### What did you expect to happen?

I expect at least import pattern 1 to work.

### How can we reproduce the issue?

index.mjs:

```js
import { PDFDocument } from "@cantoo/pdf-lib";
PDFDocument.create();
```

```json
{
  "name": "pdf-lib-sample",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "devDependencies": {
    "@cantoo/pdf-lib": "^1.19.11",
  },
  "author": "",
  "license": "ISC"
}
```

> npm i
> node index.mjs

### Version

>=1.19

### What environment are you running pdf-lib in?

Node, Other

### Checklist

- [X] My report includes a Short, Self Contained, Correct (Compilable) Example.
- [X] I have attached all PDFs, images, and other files needed to run my SSCCE.

### Additional Notes

For typescript to recognize the es/index.js import, it needs a package.json with `{ type: "module" }` in the directory at a minimum.

For node, it needs the new "exports" block in package.json

```json
  "exports": {
    ".": {
      "types": "./cjs/index.d.ts",
      "require": "./cjs/index.js",
      "import": "./es/index.js"
    }
  },
```

Additionally, the es/* directory imports need to be updated to use full file paths.

> Error [ERR_MODULE_NOT_FOUND]: Cannot find module '[redacted]/pdf-lib-sample/node_modules/@cantoo/pdf-lib/es/core/errors' imported from [redacted]/pdf-lib-sample/node_modules/@cantoo/pdf-lib/es/core/index.js

Is fixed by changing es/index.js to:

```ts
export * from './api/index.js';
export * from './core/index.js';
export * from './types/index.js';
export * from './utils/index.js';
```

But needs to be repeated all of the way down.

My current resolution is to downgrade back to 1.17, which does not have this issue.

**Comment by Sharcoux** [comment]
I'll be on it. Sorry for the delay.

**Comment by Sharcoux** [comment]
@thantos Did the prepatch fix the issue? I'll be releasing version 1.19.13 now.

**Comment by thantos** [comment]
Haven't had a chance to try yet, will try later today.

**Comment by Sharcoux** [comment]
I released version 1.19.13. I decided to embark the change, as the current version is broken anyway. Not much risk to break anything, right?

**Comment by bluepuma77** [comment]
Assuming this is related:
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'color.js' imported from /Users/.../node_modules/@cantoo/pdf-lib/es/api/colors.js
```

**Comment by Sharcoux** [comment]
Ah, indeed. Let me fix that

**Comment by Sharcoux** [comment]
Fixed in 1.19.14

**Comment by thantos** [comment]
Its working for me so far, thanks for the fix!
